exporter/awsxrayexporter/internal/translator/cause.go

```diff
@@ -486,6 +489,59 @@ func fillDotnetStacktrace(stacktrace string, exceptions []awsxray.Exception) []a
 	return exceptions
 }
 
+func fillGoStacktrace(stacktrace string, exceptions []awsxray.Exception) []awsxray.Exception {
+	var line string
+	var label string
+	var path string
+	var lineNumber int
+
+	r := textproto.NewReader(bufio.NewReader(strings.NewReader(stacktrace)))
+
+	// Skip first line containing top level exception / message
+	_, _ = r.ReadLine()
+	exception := &exceptions[0]
+	line, err := r.ReadLine()
+	if err != nil {
+		return exceptions
+	}
+
+	exception.Stack = make([]awsxray.StackFrame, 0)
+	for {
+		re := regexp.MustCompile(`^goroutine.*\brunning\b.*:$`)
```

Comment:
Generally the advice is not to use `MustCompile` as it can `panic()`, but with a constant input that is known good it should be fine here.  I would advise moving this initialization outside of the loop, possibly even outside the function declaration as the regex doesn't need to be parsed and initialized on every use.

Reply:
Yeah makes sense. but not sure it is a good idea to initialize `MustCompile` outside this function as this file contains logic of other SDK stack trace translation implementation as well and the initialization would be very specific to go stack trace translation support. I feel moving outside of the loop would be good for now.

Reply:
That's probably good enough.  We can revisit if it isn't.